### ISSUE TITLE
Support variations in "Group" names

### ISSUE THREAD

**Issue by clevengr** [icpctools/icpctools#991]
Currently the ICPC CMS provides a `groups.json` file in contest data package downloads.  The names of the "groups" come from (are I believe precisely the same as) the "sites" defined within CMS for the particular contest.  This is fine for a generic default, but it's not very flexible and causes confusion in some contests.

For example, during the North America Championship contest there was a Presentation with the title "Group Leaders" which showed the top team in each "group" (a.k.a. CMS "site").  There were numerous questions about the meaning of this "Group Leaders" presentation, because NAC doesn't have anything which the general community understands as going by the name "Group".  (In this case, there were four "groups" -- a.k.a. "sites" -- defined by CMS:  "ICPC NAC West Division", "ICPC NAC South Division", "ICPC NAC East Division", and "ICPC NAC Central Division").   However, there is nothing on the Presentation which explains the relationship between the "Group Leaders" _title_ and the fact that this particular contest had four "Divisions".  Similar issues exist in other contests, for example in the Pacific NW Regional contest where there are two "Divisions" ("D1" and "D2") spread over five physical "sites".  

We should provide an enhancement that in some way allows "overriding" the default "Group" naming with contest-specific "group names".  It should be possible, for example, to change the default title in the "Group Leaders" presentation to instead say "Division Leaders", since that is the understood "Grouping" for (in particular) the NAC.  (Note that the intention is NOT to provide support for the ability to define new groups; rather, since the "groups" are _already defined in the CMS download_, the intent is to allow changing the _title_ in the presentation to match the logical group naming associated with a particular contest.

A nice additional enhancement would be to separate the lines on the "Group Leaders" presentation and put a "group name" ahead of each group leader row.  (Currently, the presentation lists each "group leader" but it's up to the viewer to decipher which "group" a leader is a part of.  For example, how many people can look at a "Group Leaders" board showing (say) Univ. of Maryland, Univ. of Illinois, and Ohio State University and know for sure which "Group" ("Division") they are leading?  (In fact they are in three different "Divisions" -- but which ones?  Hint:  the eastern-most of them is NOT in the "East Division"!)

A further useful enhancement would be to support contests where not every "Group Leader" should be displayed on the "Group Leaders" presentation.  For example, in the PacNW contest there are TEN "groups" ("sites") defined in CMS -- two "divisions" each of which is at five physically different sites.  A "Group Leaders" display for a contest organized like this should list the "Group Leader" as the team with the highest overall rank among all teams at all physical sites that are in the same "PacNW Division".   But trying to support this may be a bit too much.

**Comment by deboer-tim** [comment]
Can you post a groups.json from NAC or PACNW? The CLICS spec has a little info to support this but not really everything you're asking for, and I can almost guarantee they weren't even configured at these contests to do what the spec allows.

In other words, I think we need the meta-discussion at the CLICS or sysops level about where the data to support this is going to come from (requires spec additions? are CCS going to support it and is anyone going to configure it?) before we talk about what the presentation shows, because there is no point building presentations if nobody is going to provide the data to say what sites or divisions exist in the first place.

**Comment by clevengr** [comment]
I completely understand that there are large complexities in some of this which certainly SHOULD be discussed in a CLICS or SysOps meeting.  However, there are others that I think could be implemented relatively cleanly and easily.  (I suppose now somebody's going to say I should split this into multiple issues... but where's the dividing line?)

As an example of what I mean: providing a way to override the presentation title "Group Leaders" with a different string such as "Division Leaders" -- couldn't that be done with a property on that specific presentation?  As another example, inserting a line above each "Group Leader" info line that contains the name of the "group" seems like something that's straightforward.  The group name would come from the existing `groups.json` group name definitions (which must already be accessible, else how is the presentation deciding which (e.g. four) teams to list as a "group leader"?)

I don't see those kinds of things as particularly hard (ok, says the guys who isn't looking at the code right now).  What I mean is, I don't really see why those kinds of things need CLICS or SysOps "meta-discussions" -- they use existing data and/or require support using existing facilities (such as property sheets).

The business about different kinds of groupings for different contests -- yes; THAT is something that bears discussion.

Here's the `groups.json` from NAC 2024:
```
[{"id":"31514","icpc_id":"31514","name":"Administrative Site"},
{"id":"31515","icpc_id":"31515","name":"NAC - NAPC Volunteers"},
{"id":"31516","icpc_id":"31516","name":"NAC Teams"},
{"id":"31517","icpc_id":"31517","name":"NAPC Teams"},
{"id":"31518","icpc_id":"31518","name":"NAC - NAPC Staff"},
{"id":"31519","icpc_id":"31519","name":"NAC - NAPC Guests"},
{"id":"35892","icpc_id":"35892","name":"NAC - Host Guests"},
{"id":"37520","icpc_id":"37520","name":"ICPC NAC South Division"},
{"id":"37521","icpc_id":"37521","name":"ICPC NAC East Division"},
{"id":"37522","icpc_id":"37522","name":"ICPC NAC West Division"},
{"id":"37523","icpc_id":"37523","name":"ICPC NAC Central Division"},
{"id":"37542","icpc_id":"37542","name":"NAC - NAPC VIPs and Sponsors"}]
```
Only groups "South", "East", "West" and "Central" had any teams actually assigned to them; the rest is Admin overhead.

Here is what I believe to be the `groups.json` that was used in the PacNW contest in Feb 2024:

```
[{"id":"31670","icpc_id":"31670","name":"Administrative Site"},
{"id":"35848","icpc_id":"35848","name":"British Columbia - D1"},
{"id":"35849","icpc_id":"35849","name":"British Columbia - D2"},
{"id":"35850","icpc_id":"35850","name":"Washington - D1"},
{"id":"35851","icpc_id":"35851","name":"Washington - D2"},
{"id":"35852","icpc_id":"35852","name":"Oregon - D1"},
{"id":"35853","icpc_id":"35853","name":"Oregon - D2"},
{"id":"35854","icpc_id":"35854","name":"California - D1"},
{"id":"35855","icpc_id":"35855","name":"California - D2"},
{"id":"35856","icpc_id":"35856","name":"Hawaii - D1"},
{"id":"35857","icpc_id":"35857","name":"Hawaii - D2"}]
```

**Comment by niemela** [comment]
The way this could (I think should) be done within the current spec goes something like this:

The NAC 2024 `groups.json` should have been:
```
[{"id":"37520","icpc_id":"37520","name":"ICPC NAC South","type":"division"},
{"id":"37521","icpc_id":"37521","name":"ICPC NAC East","type":"division"},
{"id":"37522","icpc_id":"37522","name":"ICPC NAC West","type":"division"},
{"id":"37523","icpc_id":"37523","name":"ICPC NAC Central","type":"division"}]
```

Then the presentation title could be "<type> Leaders" (in this case "Division Leaders").

You could also imagine a contest with e.g. groups of both type "division" and "site". When you have multiple kinds of groups you probably don't wont to mix them (but please provide counter-examples), so you would/could/should have presentation one per type (and possible one showing all, where the title is "Group Leaders").

**Comment by deboer-tim** [comment]
Yes, that's what I meant too. I don't think we should create something custom in the tools to define what is a site and what is a division and create presentations for those. (Also a bit of 'if we're not configuring these correctly for all clients, who is ever going to use something presentation specific')

If we can convince ourselves that these _would_ be configured correctly as per the spec, then I do think it would make sense for the tools to have e.g. a Division Leaders presentation, to allow the admin to pick and show a specific division, or a Site Leaders presentation that could be applied to all sites and would show only the local teams at each one.
